test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -3037,6 +3053,21 @@ BOOST_AUTO_TEST_CASE(gasprice)
 	ABI_CHECK(callContractFunction("f()"), encodeArgs(gasPrice()));
 }
 
+BOOST_AUTO_TEST_CASE(blockhash_single)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f() public returns (bytes32 r) {
+				r = blockhash(block.number - 1);
+			}
+		}
+	)";
+	compileAndRun(sourceCode);
+	u256 blockNumber = m_blockNumber;
+	auto hash = blockHash(blockNumber - 1);
```

Comment:
Previously this would have had to be ``blockHash(blockNumber)`` instead of ``blockHash(blockNumber - 1)``.